fondant/compiler.py

```diff
@@ -126,10 +126,17 @@ def _generate_spec(self, pipeline: Pipeline) -> dict:
             volumes = [asdict(volume)] if volume else []
 
             services[safe_component_name] = {
-                "image": component_op.component_spec.image,
                 "command": command,
                 "depends_on": depends_on,
                 "volumes": volumes,
```

Comment:
only when the base_path is local otherwise the volume will be empty
```
    depends_on:
      second_component:
        condition: service_completed_successfully
    volumes: []
```